### PR TITLE
Add MDX type: text/mdx

### PR DESCRIPTION
**Pull request by silvenon**
Continuing from #136.

### CODE REVIEW

File: src/custom-types.json [hunk 1]
```diff
@@ -743,6 +743,14 @@
       "https://tools.ietf.org/html/rfc7763"
     ]
   },
+  "text/mdx": {
+    "compressible": false,
+    "extensions": ["mdx"],
```
Comment (dougwilson) [inline src/custom-types.json]: Where in https://github.com/mdx-js/specification is the extension `.mdx` stated, or is there another doc that states the extension map?

Comment (silvenon) [reply src/custom-types.json]: I didn't realize it doesn't specify file extension, I'm sorry. (I'm nailing these mime type PRs, aren't I? 😓) I opened a PR to add the extension to the spec, the team is super responsive now, so it should be a matter of a day or two before it gets reviewed. I'll ping you once it's resolved.

Comment (dougwilson) [reply src/custom-types.json]: It's no problem. That's mainly why the recommended way to add new types and stuff is https://github.com/jshttp/mime-db#adding-custom-media-types and not through PRs; there is a form there that asks for all the required information.

**Comment by silvenon** [comment]
@dougwilson merged! So this PR should be ready 😄

File: src/custom-types.json [hunk 2]
```diff
@@ -743,6 +743,14 @@
       "https://tools.ietf.org/html/rfc7763"
     ]
   },
+  "text/mdx": {
+    "compressible": false,
```
Comment (dougwilson) [inline src/custom-types.json]: Isn't mdx a text file? This is just a comment, but wanted to not this incase it was a mistake

**Comment by dougwilson** [comment]
Nice!

File: src/custom-types.json (hunk 2 above)
Comment (silvenon) [reply src/custom-types.json]: Yeah, not sure what I was thinking, fixed to `true`.

**Pull request merged**